text/0517-io-os-reform.md

```diff
@@ -556,6 +556,15 @@ buffers as input. This has multiple benefits:
   results are wanted, one can use `read_to_end` and convert to a
   `String` only at the end.
 
+`read_to_end` and `read_to_string` return a `Result` with either the
+number of bytes read or an error.  If a short read occurs, that call
+gives the (short) data and returns the number of bytes written into
+that buffer.  If the error was non-transient, future calls shall return
+the reason for the error (usually `EndOfFile`).
```

Comment:
Point taken.  Honestly, perhaps it would be better just to remove these methods and let them get handled at the user level.  I would certainly be more in favor of that than standardizing something that was not vetted.

Anyway, the thought was that sometimes you might intend to read "everything that's there".  Depending on how things are set up, those (system call) reads (i.e. Rust's implementation) might get some partial bytes and not actually block.  Without this clause, the implementation would need to busy wait (or "busy read" i guess) to deal with non-blocking and non-terminating sources.

I would be much more in favor of dropping these from the trait and only providing them as specific methods on types that make sense (i.e. synchronous files).